examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -344,7 +345,7 @@ class VideoDemo extends StatefulWidget {
 final DeviceInfoPlugin deviceInfoPlugin = DeviceInfoPlugin();
 
 Future<bool> isIOSSimulator() async {
-  return Platform.isIOS && !(await deviceInfoPlugin.iosInfo).isPhysicalDevice;
+  return defaultTargetPlatform == TargetPlatform.iOS && !(await deviceInfoPlugin.iosInfo).isPhysicalDevice;
```

Comment:
If we switch to TargetPlatform, this should be a switch.

Reply:
default Target Platform is wrong here. It does not reliably say if you're on the iOS simulator: it just says what your preferred UI style is. you can toggle it by pressing "o" at the console in flutter run, for instance.

Reply:
Ahh yes.  But we could still expose something like `TargetPlatform currentPlatform` somewhere and have that delegate to Platform when io is available